python-frontend/src/main/java/org/sonar/plugins/python/api/PythonFile.java

```diff
@@ -29,4 +29,6 @@
 
   URI uri();
 
+  PythonVersion pythonVersion();
```

Comment:
Because we need PythonVersion to be available at least in SymbolTableBuilder and in Checks classes, I wanted to avoid passing another constructor paramer to SymbolTableBuilder.
Note that PythonVersion will be created only once per project, each PythonFile will reference the same object. What do you think? Do you believe it should be put somewhere else?

Reply:
My only concern is that this API implies that a project could contain files for python 2 and files for python 3. Would we be able to support that? If a built-in such as `str` doesn't mean exactly the same thing for file1.py and for file2.py, what if one of those files depends on the other one? Will we have to handle some unnecessary complexity?

Reply:
Indeed, that's a good point. I agree that we shouldn't keep it as PythonFile level.

Reply:
After c6c90b5, I don't see how the implementations of rules will have access to PythonVersion.

Reply:
At the moment, I think we only need it in SymbolTableBuilder, so I would suggest to make it accessible to rules as soon as we need it